File: app/src/main/java/com/beibei/init/common/networkJava/helper/RetrofitHelper.java
package com.beibei.init.common.networkJava.helper;

import com.beibei.init.common.constant.Constant;
import com.joyou.smartcity.common.newNetwork.helper.OkHttpClientHelper;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitHelper {

    private OkHttpClient mClient = OkHttpClientHelper.Companion.getInstance().getOkHttpClient();

    private static volatile RetrofitHelper retrofitHelper = null;

    private RetrofitHelper() {
    }

    private static RetrofitHelper getInstance() {
        if (retrofitHelper == null) {
            synchronized (RetrofitHelper.class) {
                if (retrofitHelper == null) {
                    retrofitHelper = new RetrofitHelper();
                }
            }
        }
        return retrofitHelper;
    }

    private Retrofit retrofit = null;

    private Retrofit getRetrofit() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(Constant.HOST) // 设置 网络请求 Url
                    .addConverterFactory(GsonConverterFactory.create()) //设置使用Gson解析(记得加入依赖)
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create()) // 支持RxJava
                    .client(mClient)   //关联自定义的OkHttp
                    .build();
        }
        return retrofit;
    }

    public static <T> T createRequest(Class<T> requestInterfaceClass) {
        return RetrofitHelper.getInstance()
                .getRetrofit()
                .create(requestInterfaceClass);
    }
}

File: app/src/main/java/com/beibei/init/common/network/helper/RetrofitHelper.kt
package com.beibei.init.common.newNetwork.helper

import com.beibei.init.common.constant.Constant
import com.beibei.init.common.newNetwork.convert.MyGsonConverterFactory
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory

/**
 * Created by beibei on 2018/3/19.
 */

class RetrofitHelper private constructor() {

    private val mClient: OkHttpClient = OkHttpClientHelper.instance.okHttpClient
    private var mRetrofit: Retrofit? = null

    //步骤4：创建Retrofit对象
    fun getRetrofit(): Retrofit {
        if (mRetrofit == null) {
            mRetrofit = Retrofit.Builder()
                    .baseUrl(Constant.HOST)
                    .addConverterFactory(MyGsonConverterFactory.create())  //添加Gson支持
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())  //添加RxJava支持
                    .client(mClient)                                            //关联okhttp
                    .build()
        }
        return mRetrofit!!
    }

    companion object {
        private var helper: RetrofitHelper? = null

        //单例 保证对象唯一
        val instance: RetrofitHelper
            get() {
                if (helper == null) {
                    synchronized(RetrofitHelper::class.java) {
                        if (helper == null) {
                            helper = RetrofitHelper()
                        }
                    }
                }
                return helper!!
            }

        // 步骤5：创建 网络请求接口 的实例
        fun <T> createApiRequest(clazz: Class<T>): T {
            return RetrofitHelper.instance
                    .getRetrofit()
                    .create(clazz)
        }
    }
}